src/runtime/audience-action-flow-test.js

```diff
@@ -445,10 +508,38 @@ describes.realWin('AudienceActionFlow', {}, (env) => {
       .withExactArgs(successSurveyDataTransferResponse)
       .once();
 
-    const surveyDataTransferRequest = new SurveyDataTransferRequest();
-    const messageCallback = messageMap[surveyDataTransferRequest.label()];
-    messageCallback(surveyDataTransferRequest);
+    const messageCallback = messageMap[TEST_SURVEYDATATRANSFERREQUEST.label()];
+    messageCallback(TEST_SURVEYDATATRANSFERREQUEST);
+
+    winMock.verify();
+    activityIframeViewMock.verify();
+  });
+
+  it(`handles a SurveyDataTransferRequest with failed logging`, async () => {
+    setGtagToNonfunction();
```

Comment:
(Optional) consider having the implementation within the test if the function is going to be used only for this.

Reply:
I created a function for this firstly for clarity, so that the env setup is separate from the test and with the other setup as well, and secondly so this can be reused / refactored for other types of failure tests.